port_ocean/context/event.py

```diff
@@ -50,6 +56,37 @@ class EventContext:
     _parent_event: Optional["EventContext"] = None
     _event_id: str = field(default_factory=lambda: str(uuid4()))
     _on_abort_callbacks: list[AbortCallbackFunction] = field(default_factory=list)
+    _failed_entity_callback_list: list[
+        Tuple[Entity, Callable[[], Coroutine[Any, Any, Entity | Literal[False] | None]]]
+    ] = field(default_factory=list)
+
+    def register_failed_upsert_call_arguments(
+        self,
+        entity: Entity,
+        func: Callable[[], Coroutine[Any, Any, Entity | Literal[False] | None]],
+    ) -> None:
+        self._failed_entity_callback_list.append((entity, func))
+
+    async def handle_failed(self) -> None:
+        entity_map: dict[
+            str, Callable[[], Coroutine[Any, Any, Entity | Literal[False] | None]]
+        ] = {
+            f"{obj.identifier}-{obj.blueprint}": func
+            for obj, func in self._failed_entity_callback_list
+        }
+        entity_list: list[Entity] = [
+            obj for obj, func in self._failed_entity_callback_list
+        ]
```

Comment:
I wonder if this is actually a method that should be in the `event`(and maybe the whole handling stuff), I am leaning towards having a class for managing all of this and moving the instance for it to the event instance.